lib/ansible/modules/system/mdadm.py

```diff
@@ -0,0 +1,516 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Anthony ARNAUD <anthony.arnaud@corp.ovh.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+author:
+    - Anthony ARNAUD  (@bluelogon)
+module: mdadm
+short_description: Configure software RAID devices
+description:
+  - This module creates, removes software RAID devices.
+version_added: "2.6"
+options:
+  name:
+    description:
+    - Array name.
+    required: true
+  devices:
+    description:
+    - Define disk devices to assign to array.
+  raid_devices:
+    description:
+    - Specify the number of active devices in the array. Default is the total devices
+  level:
+    description:
+    - Define the array raid level
+    choices: ['linear','stripe','raid0','raid1','raid4','raid5','raid6','raid10','multipath','faulty','container']
+    default: "raid10"
+  metadata:
+    description:
+    - Declare the style of RAID metadata (superblock) to be used
+    choices: ['0','0.90','1.0','1.1','1.2']
+    default: "1.2"
+  state:
+    description:
+    - Control if the MD exists. If C(present) and the
+      MD does not already exist then the C(devices) option is required.
+    choices: [ absent, present, info ]
+    default: present
+  opts:
+    description:
+    - Free-form options to be passed to the mdadm command at creation.
+'''
+
+EXAMPLES = '''
+- name: Create a RAID1 /dev/md/myraid1
+  mdadm:
+    name: myraid1
+    devices:
+      - /dev/sda1
+      - /dev/sdb1
+    level: raid1
+    state: present
+
+- name: Create a RAID5 with on spare
+  mdadm:
+    name: 0
+    devices:
+      - /dev/sda1
+      - /dev/sdb1
+      - /dev/sdc1
+      - /dev/sdd1
+    level: raid5
+    raid_devices: 3
+    state: present
+
+- name: Destroy /dev/md0
+  mdadm:
+    name: 0
+    state: absent
+'''
+
+RETURN = '''
+changed:
+  description: Return changed for mdadm actions as true or false.
+  returned: always
+  type: boolean
+  version_added: 2.6
```

Comment:
Not necessary to add "version_added" to all return values because it is a new module.
Only need for "new" return values above 2.6